ivy/functional/backends/numpy/general.py

```diff
@@ -341,7 +341,12 @@ def scatter_nd(
 scatter_nd.support_native_out = True
 
 
-def shape(x: np.ndarray, /, *, as_array: bool = False) -> Union[ivy.Shape, ivy.Array]:
+def shape(
+    x: np.ndarray,
+    /,
+    *,
+    as_array: Optional[bool] = False,
```

Comment:
The `Optional` type hint is for arguments allowing a `None` value, here only bool is allowed so the original type hint is correct, the same comment applies to the other files below! :)